src/main/java/trackerbot/command/Command.java

```diff
@@ -0,0 +1,226 @@
+     * @throws TrackerBotException The internal TrackerBotException, for expected and recoverable
+     *                             errors to display on the UI.
+     */
+    public abstract void execute(TaskList tasks, Ui ui) throws TrackerBotException;
+
+    public abstract boolean isExit();
+
+    /**
+     * Factory method for Command.
+     * <p>Depending on the keyword passed in, the method will generate an
+     * appropriate instance of a subtype of Command. Currently, this method can generate
+     * these subtypes:</p>
+     * <ul>
+     *     <li>AddCommand, to add Tasks into the Tracker.</li>
+     *     <li>DeleteCommand, to remove Tasks from the Tracker.</li>
+     *     <li>ToggleCommand, to mark/unmark Tasks.</li>
+     *     <li>FindCommand, to find all Tasks matching a substring.</li>
+     *     <li>ListCommand, to display Tasks in the Tracker.</li>
+     *     <li>ExitCommand, to tell the Tracker to exit.</li>
+     *     <li>UnknownCommand, which will throw an error on execute.</li>
+     * </ul>
+     * @param keyword The keyword passed in by the user input.
+     * @param commandField The description of the user input.
+     * @return Some subtype of Command related to keyword.
+     */
+    public static Command of(String keyword, String commandField) {
+        CommandType parsedType = CommandType.UNKNOWN;
+        for (CommandType command: CommandType.values()) {
+            if (keyword.equals(command.getKeyword())) {
+                parsedType = command;
+                break;
+            }
+        }
+
+        Command result;
+        switch (parsedType) {
+        case TODO:
+            // Fallthrough
+        case DEADLINE:
+            // Fallthrough
+        case EVENT:
+            result = new AddCommand(parsedType, commandField);
+            break;
+        case DELETE:
+            result = new DeleteCommand(commandField);
+            break;
+        case MARK:
+            // Fallthrough
+        case UNMARK:
+            result = new ToggleCommand(parsedType, commandField);
+            break;
+        case FIND:
+            result = new FindCommand(commandField);
+            break;
+        case LIST:
+            result = new ListCommand();
+            break;
+        case BYE:
+            result = new ExitCommand();
+            break;
+        default:
+            // this handles the UNKNOWN case
+            result = new UnknownCommand();
+        }
+        return result;
+    }
+
+    private static class AddCommand extends Command {
+        private final String commandFields;
+        private final CommandType type;
+
+        private AddCommand(CommandType type, String commandFields) {
+            this.commandFields = commandFields;
+            this.type = type;
+        }
+
+        public void execute(TaskList tasks, Ui ui) throws TrackerBotException {
+            ui.showMessage(tasks.add(type, commandFields));
+        }
+
+        public boolean isExit() {
```

Comment:
I noticed that the `@Override` notation for `execute` and `isExit` function is not included for all the child classes. It would be good to include this notation so that any errors can be caught by the compiler at compile time!

Reply:
Ah, I must have forgotten about this. Thanks for pointing it out!